server/src/main/java/com/linecorp/line/auth/fido/fido2/server/service/ResponseServiceImpl.java

```diff
@@ -300,13 +300,14 @@ protected void verifySignature(ServerAuthPublicKeyCredential serverPublicKeyCred
         byte[] signatureBytes = Base64.getUrlDecoder().decode(serverPublicKeyCredential.getResponse().getSignature());
         boolean result = SignatureHelper.verifySignature(userKey.getPublicKey(), toBeSignedMessage, signatureBytes, userKey.getAlgorithm());
         if (!result) {
-            throw new FIDO2ServerRuntimeException(InternalErrorCode.ASSERTION_SIGNATURE_VERIFICATION_FAIL);
+            throw FIDO2ServerRuntimeException.makeSignatureVerificationError(userKey.getAaguid());
         }
     }
 
-    protected VerifyCredentialResult createVerifyCredentialResult(AuthenticatorData authData, UserKey userKey) {
+    protected VerifyCredentialResult createVerifyCredentialResult(AuthenticatorData authData, UserKey userKey, String aaguid) {
         return VerifyCredentialResult
                 .builder()
+                .aaguid(aaguid)
```

Comment:
(nit) Since the actual aaguid is got from `userKey`, I think we can keep the method simple by using its getter.

```suggestion
    protected VerifyCredentialResult createVerifyCredentialResult(AuthenticatorData authData, UserKey userKey) {
        return VerifyCredentialResult
                .builder()
                .aaguid(userKey.getAaguid())
```